core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -346,19 +349,15 @@ private InputStream getUncompressedBody() {
   }
 
   private InputStream getCompressedBody() {
-    if (compression == Compression.NONE) {
+    if (decompressor == MessageEncoding.NONE) {
       throw Status.INTERNAL.withDescription(
           "Can't decode compressed frame as compression not configured.").asRuntimeException();
     }
 
-    if (compression != Compression.GZIP) {
-      throw new AssertionError("Unknown compression type");
-    }
-
     try {
-      return new GZIPInputStream(ReadableBuffers.openStream(nextFrame, true));
+      return decompressor.decompress(ReadableBuffers.openStream(nextFrame, true));
     } catch (IOException e) {
-      throw new RuntimeException(e);
+      throw Throwables.propagate(e);
```

Comment:
I'm actually trying to remove those, instead of adding more. I've had very good success with looking for Throwables.propagate() to find exception handling bugs; the tool itself isn't the problem, but its overuse has led to ignoring errors.

Still, is it really "consistency"? Can you find a place we catch a `Throwable` other than `Throwable` or `Exception` and use `Throwables.propagate()`? Those are the only cases where `Throwables.propagate()` seems useful, in order to prevent needless exception-wrapping.